test/e2e/k8s/operator.go

```diff
@@ -205,31 +202,71 @@ func mergeConfiguration(initCfg *Config) *Config {
 	return config
 }
 
-func initCustomZapLogger(level, encoding, logFileName string) (*zap.Logger, error) {
-	lv := zap.AtomicLevel{}
-	err := lv.UnmarshalText([]byte(strings.ToLower(level)))
-	if err != nil {
-		return nil, err
+type ManagerStart func(ctx context.Context) error
+type ManagerConfig func(config *Config)
+
+func managerDefaults() *Config {
+	return &Config{
+		AtlasDomain:                 "https://cloud-qa.mongodb.com/",
+		EnableLeaderElection:        false,
+		MetricsAddr:                 ":8080",
+		Namespace:                   "mongodb-atlas-system",
+		WatchedNamespaces:           map[string]bool{},
+		ProbeAddr:                   ":8081",
+		GlobalAPISecret:             client.ObjectKey{},
+		ObjectDeletionProtection:    false,
+		SubObjectDeletionProtection: false,
 	}
+}
+
+func WithNamespaces(namespaces ...string) ManagerConfig {
+	return func(config *Config) {
+		for _, namespace := range namespaces {
+			config.WatchedNamespaces[namespace] = true
+		}
+
+		if len(namespaces) == 1 {
+			config.Namespace = namespaces[0]
+		}
+	}
+}
+
+func WithObjectDeletionProtection(flag bool) ManagerConfig {
+	return func(config *Config) {
+		config.ObjectDeletionProtection = flag
+	}
+}
 
-	enc := strings.ToLower(encoding)
-	if enc != "json" && enc != "console" {
-		return nil, errors.New("'encoding' parameter can only by either 'json' or 'console'")
+func WithSubObjectDeletionProtection(flag bool) ManagerConfig {
+	return func(config *Config) {
+		config.SubObjectDeletionProtection = flag
 	}
+}
 
-	cfg := zap.Config{
-		Level:             lv,
-		OutputPaths:       []string{logFileName},
-		DisableCaller:     false,
-		DisableStacktrace: false,
-		Encoding:          enc,
-		EncoderConfig: zapcore.EncoderConfig{
-			MessageKey:  "msg",
-			LevelKey:    "level",
-			EncodeLevel: zapcore.CapitalLevelEncoder,
-			TimeKey:     "time",
-			EncodeTime:  zapcore.ISO8601TimeEncoder,
-		},
+func WithGlobalKey(key client.ObjectKey) ManagerConfig {
+	return func(config *Config) {
+		config.GlobalAPISecret = key
 	}
-	return cfg.Build()
+}
+
+func RunManager(withConfigs ...ManagerConfig) (ManagerStart, error) {
```

Comment:
aha! not sure where this run was happening before though

Reply:
got it now, the run before was not a run but a build, so the name change makes sense. The manager is being run just the same as before.